data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/OnVoltageConstraintImpl.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.farao_community.farao.data.crac_impl;
+
+import com.farao_community.farao.data.crac_api.Instant;
+import com.farao_community.farao.data.crac_api.State;
+import com.farao_community.farao.data.crac_api.cnec.VoltageCnec;
+import com.farao_community.farao.data.crac_api.usage_rule.OnVoltageConstraint;
+import com.farao_community.farao.data.crac_api.usage_rule.UsageMethod;
+
+/**
+ * @author Fabrice Buscaylet {@literal <fabrice.buscaylet at artelys.com>}
+ */
+public class OnVoltageConstraintImpl extends AbstractUsageRule implements OnVoltageConstraint {
+    private final Instant instant;
+
+    private final VoltageCnec voltageCnec;
+
+    OnVoltageConstraintImpl(Instant instant, VoltageCnec angleCnec) {
+        super(UsageMethod.TO_BE_EVALUATED);
+        this.instant = instant;
+        this.voltageCnec = angleCnec;
+    }
+
+    @Override
+    public VoltageCnec getVoltageCnec() {
+        return voltageCnec;
+    }
+
+    @Override
+    public Instant getInstant() {
+        return instant;
+    }
+
+    @Override
+    public UsageMethod getUsageMethod(State state) {
+        return UsageMethod.TO_BE_EVALUATED;
```

Comment:
In fact, we would like to have the same behavior here as in OnAngleConstaintImpl (not mentioned in the US). That's because it does not make sense to have a state that differs from the voltageCnec state.
See the code below : 

```java
@Override
public UsageMethod getUsageMethod(State state) {
    if (state.isPreventive()) {
        return state.getInstant().equals(instant) ? UsageMethod.TO_BE_EVALUATED : UsageMethod.UNDEFINED;
    } else {
        return state.getInstant().equals(instant) && state.equals(this.angleCnec.getState()) ? UsageMethod.TO_BE_EVALUATED : UsageMethod.UNDEFINED;
    }
}


```

Reply:
I will do as you said but  I'm not sure the CSA 25 User story is up to date. It will need an update to reflect the code behavior